### PR TITLE
feat(linter) eslint plugin unicorn: no useless spread

### PR DESCRIPTION
**Pull request by camc314**
Tasks:
 - [x] Add approprite error message
 - [x] clean up code

### CODE REVIEW

File: crates/oxc_linter/src/snapshots/no_useless_spread.snap [hunk 1]
```diff
@@ -0,0 +1,677 @@
+---
+source: crates/oxc_linter/src/tester.rs
+expression: no_useless_spread
+---
+  ⚠ eslint-plugin-unicorn(no-useless-spread): Using a spread operator here creates a new array unnecessarily.
+   ╭─[no_useless_spread.tsx:1:1]
+ 1 │ const array = [...[a]]
+   ·                   ───
```
Comment (Boshen) [inline crates/oxc_linter/src/snapshots/no_useless_spread.snap:8]: We don't have a span to `...`, but you can probably get it from the first 3 chars of the `SpreadElement`.

**Review by Boshen — COMMENTED** [review]
Just a nit pick on the span

File: crates/oxc_linter/src/snapshots/no_useless_spread.snap (hunk 1 above)
Comment (camc314) [reply crates/oxc_linter/src/snapshots/no_useless_spread.snap]: good call

**Comment by Boshen** [comment]
Nice! 1 more correctness rule.

**Pull request merged**